File: appinventor/components-ios/src/ButtonShape.swift
// -*- mode: swift; swift-mode:basic-offset: 2; -*-
// Copyright 2026 MIT, All rights reserved
// Released under the Apache License, Version 2.0
// https://www.apache.org/licenses/LICENSE-2.0

@objc public class ButtonShape: NSObject, OptionList {
  @objc public static let Default = ButtonShape(0)
  @objc public static let Rounded = ButtonShape(1)
  @objc public static let Rectangular = ButtonShape(2)
  @objc public static let Oval = ButtonShape(3)

  private static let LOOKUP: [Int32: ButtonShape] = generateOptionsLookup(Default, Rounded,
      Rectangular, Oval)

  let value: Int32

  @objc private init(_ value: Int32) {
    self.value = value
  }

  @objc public func toUnderlyingValue() -> AnyObject {
    return value as AnyObject
  }

  @objc public static func fromUnderlyingValue(_ value: Int32) -> ButtonShape? {
    return LOOKUP[value]
  }
}

File: appinventor/components/src/com/google/appinventor/components/common/ButtonShape.java
// -*- mode: java; c-basic-offset: 2; -*-
// Copyright 2026 MIT, All rights reserved
// Released under the Apache License, Version 2.0
// http://www.apache.org/licenses/LICENSE-2.0

package com.google.appinventor.components.common;

import java.util.HashMap;
import java.util.Map;

/**
 * Defines the shapes supported by button-like components.
 */
public enum ButtonShape implements OptionList<Integer> {
  @Default
  Default(0),
  Rounded(1),
  Rectangular(2),
  Oval(3);

  private final int value;

  ButtonShape(int value) {
    this.value = value;
  }

  public Integer toUnderlyingValue() {
    return value;
  }

  private static final Map<Integer, ButtonShape> lookup = new HashMap<>();

  static {
    for (ButtonShape shape : ButtonShape.values()) {
      lookup.put(shape.toUnderlyingValue(), shape);
    }
  }

  public static ButtonShape fromUnderlyingValue(Integer shape) {
    return lookup.get(shape);
  }
}
